integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/mail/EduMailSender.java

```diff
@@ -86,10 +98,54 @@ protected PasswordAuthentication getPasswordAuthentication() {
             mimeMultipart.addBodyPart(mimeBodyPart);
             message.setContent(mimeMultipart);
 
+            Transport transport = session.getTransport("smtps");
+
+            Long maxMessageSize = getMaxMessageSize(transport);
+
+            if (maxMessageSize != null) {
+                long messageSize = getMessageSize(message);
+
+                if (messageSize > maxMessageSize) {
+                    String conversationId = request.getEnvelope().getConversationId();
+
+                    if (conversationIds.add(conversationId)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS134)